test/test-lib/fake-device.ts

```diff
@@ -0,0 +1,56 @@
+import 'mocha';
+import { expect } from 'chai';
+import * as uuid from 'node-uuid';
+import * as randomstring from 'randomstring';
+
+import { app } from '../../init';
+
+import supertest = require('./supertest');
+
+export async function provisionDevice(admin: string, appId: number) {
+	const deviceType = await supertest(app, admin)
+		.get(`/resin/application(${appId})?$expand=is_for__device_type`)
+		.expect(200)
+		.then(res => {
+			expect(res.body)
+				.to.have.property('d')
+				.that.is.an('array');
+			expect(res.body.d).to.have.lengthOf(
+				1,
+				`Incorrect number of applications found for ID ${appId}`,
+			);
+			expect(res.body.d[0]).to.have.property('is_for__device_type');
+			expect(res.body.d[0].is_for__device_type)
+				.to.be.an('array')
+				.with.lengthOf(1);
+			expect(res.body.d[0].is_for__device_type[0]).to.have.property('slug');
+
+			return res.body.d[0].is_for__device_type[0].slug as string;
+		});
+
+	const device = await supertest(app, admin)
```

Comment:
```suggestion
	const { body: device } = await supertest(app, admin)
```
and drop the `.then(res => res.body)`